### PR TITLE
fix(refresher): prevent clearing virtual scroll overflow styling

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

`ion-refresher` will overwrite 3rd-party overflow styles assigned by virtual scroll libraries (e.g. Virtuoso). 

Issue URL: #26553

## What is the new behavior?

- `ion-refresher` will only assign `hidden` overflow styling when overflow is visible.
- `ion-refresher` will restore inline overflow styles set on the scroll element.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Dev-build: `6.4.3-dev.11674237446.1a3aa9dc`

Updated reproduction: https://stackblitz.com/edit/angular-4uocfn-n1qkbm?file=src%2Fmain.tsx,package.json

Also the local scroll target test template has been updated to leverage the underlying behavior that originally causes this bug.

### CODE REVIEW

File: core/src/components/refresher/refresher.tsx [hunk 1]
```diff
@@ -729,7 +729,15 @@ export class Refresher implements ComponentInterface {
...
         scrollStyle.transitionDuration = backgroundStyle.transitionDuration = duration;
         scrollStyle.transitionDelay = backgroundStyle.transitionDelay = delay;
-        scrollStyle.overflow = overflowVisible ? 'hidden' : '';
+        /**
+         * The goal is to avoid overriding the settings of certain virtual scroll
+         * libraries that already set the overflow style on the scroll container.
+         */
+        if (overflowVisible) {
+          scrollStyle.overflow = 'hidden';
+        } else if (scrollStyle.overflow === 'hidden') {
```
Comment (liamdebeasi) [inline core/src/components/refresher/refresher.tsx:738]: I don't understand how this fixes the issue. I tried the reproduction you provided in https://github.com/ionic-team/ionic-framework/issues/26553#issuecomment-1385754385 with iOS mode, and the scroller is still disabled after completing the pull to refresh action.

My understanding of the issue is that we are overriding an `overflow` style that is set inline on the scroller (as opposed to in a stylesheet). If that's the case, we likely need to keep track of the inline style before we call `setCss` so we can restore the correct style.

File: core/src/components/refresher/refresher.tsx [hunk 2]
```diff
@@ -729,11 +731,35 @@ export class Refresher implements ComponentInterface {
...
+    }
+  }
+
+  private restoreOverflowStyle() {
+    if (this.overflowStyles !== undefined && this.scrollEl !== undefined) {
+      const { overflow, overflowX, overflowY } = this.overflowStyles;
+      this.scrollEl.style.overflow = overflow;
+      this.scrollEl.style.overflowX = overflowX;
+      this.scrollEl.style.overflowY = overflowY;
+    }
```
Comment (liamdebeasi) [inline core/src/components/refresher/refresher.tsx:760]: Do we need to clear `this.overflowStyles` when done?

File: core/src/components/refresher/refresher.tsx [hunk 3]
```diff
@@ -729,11 +731,35 @@ export class Refresher implements ComponentInterface {
         scrollStyle.transform = backgroundStyle.transform = y > 0 ? `translateY(${y}px) translateZ(0px)` : '';
         scrollStyle.transitionDuration = backgroundStyle.transitionDuration = duration;
         scrollStyle.transitionDelay = backgroundStyle.transitionDelay = delay;
-        scrollStyle.overflow = overflowVisible ? 'hidden' : '';
```
Comment (liamdebeasi) [inline core/src/components/refresher/refresher.tsx:732]: One thing I noticed with the MD native refresher is that it sets `--overflow: hidden` (assumes `ion-content`) so the overflow is never set to `hidden` on virtual scroll targets. Not something we need to fix in this PR, but should probably track this.

**Review by liamdebeasi — APPROVED** [review]
A couple comments, but otherwise looks good. Great job!

File: core/src/components/refresher/refresher.tsx (hunk 2 above)
Comment (sean-perkins) [reply core/src/components/refresher/refresher.tsx]: I did some initial testing and clearing the overflow styles does not impact the experience, so I updated the implementation to clear the styles after assigning them.

**Pull request merged**